lib/vmnet.ml

```diff
@@ -70,41 +74,81 @@ let error_of_int =
   | 1008 -> Too_many_packets
   | err  -> Unknown err
 
+type ipv4_config = {
+    ipv4_start_address: Ipaddr_sexp.V4.t;
+    ipv4_end_address: Ipaddr_sexp.V4.t;
+    ipv4_netmask: Ipaddr_sexp.V4.t;
+} [@@deriving sexp]
+
 type mode =
   | Host_mode
   | Shared_mode
   | Bridged_mode of string [@@deriving sexp]
 
+type proto =
+  | TCP
+  | UDP
+  | ICMP
+  | Other of int [@@deriving sexp]
+
+let int_of_proto =
+  function
+  | TCP     -> 6
+  | UDP     -> 17
+  | ICMP    -> 1
+  | Other x -> x
+
+let proto_of_int =
+  function
+  | 6  -> TCP
+  | 17 -> UDP
+  | 1  -> ICMP
+  | x  -> Other x
+
 type t = {
   iface: interface_ref sexp_opaque;
   name: string;
   mtu: int;
   mac: Macaddr_sexp.t;
   max_packet_size: int;
+  uuid: Uuidm.t sexp_opaque;
 } [@@deriving sexp_of]
 
 let mac {mac; _} = mac
 let mtu {mtu; _} = mtu
 let max_packet_size {max_packet_size; _} = max_packet_size
+let uuid {uuid; _} = uuid
 
 let iface_num = ref 0
 
-let init ?(mode = Shared_mode) () =
+let init ?(mode = Shared_mode) ?(uuid = Uuidm.nil) ?ipv4_config () =
   let mode, iface =
     match mode with
     | Host_mode -> (1000, "")
     | Shared_mode -> (1001, "")
     | Bridged_mode iface -> (1002, iface)
   in
+  let ip_to_str ip = Ipaddr.V4.to_string ip in
+  let ipv4_start_address, ipv4_end_address, ipv4_netmask =
```

Comment:
This seems like it could be expressed as a single tuple of `Some (start, end, netmask)` instead of individual options. That would in turn simplify the C bindings and remove the need for CAMLxparam (due to having >5 arguments).  I think it can be merged like this, but it would be simpler to do that and have just one FFI call instead of the bytecode/native alternatives.

Reply:
Yeah agreed, I wasn't sure if it would work to pass in an optional field, but seems it works fine! Updated the PR